docs/Testing.md

```diff
@@ -0,0 +1,132 @@
+# Testing
+
+A test case takes the form of a binary \<basename\>.wasm file, next to that
+that there will always be a \<basename\>.\<ext\> source file from which the
+test was originally compiled from which can be used as a reference in the event
+of an error.
+
+Additionally, any of the following optional auxilary files and directories may
+be present:
+- `<basename>.arg`
+- `<basename>.env`
+- `<basename>.dir`
+- `<basename>.stdin`
+- `<basename>.stdout`
+- `<basename>.stderr`
+- `<basename>.status`
+
+## Running tests
+
+To run a test do the following steps to prepare for execution:
+
+- Prepare inputs
+  - Given an `<input>.wasm` file; take the `<basename>` of said file.
+  - If `<basename>.<arg>` exists; take the program arguments from said file.
```

Comment:
It's a minor detail, but I'd find this more clear with ".args" instead of ".arg".

Also, we should document the whitespacing convention in the arguments file. Can we say that arguments are separated by whitespace, and '"' quotes may be used to enclose an argument containing whitespace?